src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotUseEndOfStreamInAsyncMethodsTests.cs

```diff
@@ -0,0 +1,490 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DoNotUseEndOfStreamInAsyncMethodsAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DoNotUseEndOfStreamInAsyncMethodsAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class DoNotUseEndOfStreamInAsyncMethodsTests
+    {
+        [Fact]
+        public async Task AsyncMethod_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System.IO;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    private StreamReader _field;
+
+                    public StreamReader Property { get; set; }
+
+                    async Task M(StreamReader parameter, Stream stream)
+                    {
+                        var local = new StreamReader(stream);
+
+                        _ = [|local.EndOfStream|];
+                        _ = [|parameter.EndOfStream|];
+                        _ = [|_field.EndOfStream|];
+                        _ = [|Property.EndOfStream|];
```

Comment:
These should include the diagnostic ID, e.g.
```C#
_ = [|CA2024:Property.EndOfStream|];
```

Reply:
I've omitted the diagnostic ID in the test markup in the past for analyzers that only use a single ID, but I can add it for clarity if you want.

Reply:
Thanks. I think it's helpful for clarity and for specificity: if another diagnostic were to trigger here, the test wouldn't be able to distinguish.

Reply:
I've added the diagnostic ID to the markup!